pkg/storage/s3.go

```diff
@@ -279,8 +279,10 @@ func newS3Storage(backend *backup.S3, opts *ExternalStorageOptions) (*S3Storage,
 			return nil, errors.Annotatef(berrors.ErrStorageInvalidConfig, "Bucket %s is not accessible: %v", qs.Bucket, err)
 		}
 	}
+	if len(qs.Prefix) > 0 {
+		qs.Prefix += "/"
```

Comment:
What if there is a '/' already?

Reply:
what if we just [`path.Clean`](https://pkg.go.dev/path/#Clean) it 🤷

Reply:
The `Prefix` is parsed from url it can't contains the "/" suffix. See: https://github.com/pingcap/br/blob/578be7fb516599dc2c6aca2bfeec73cb9c378943/pkg/storage/parse.go#L69-L70

Reply:
We just not want the bare "/" prefix. for other non-empty prefixes,  the suffix "/" is needed. 
BTW, maybe we shouldn't depend on the `Prefix` filed to contain the "/" suffix, but in this way, we need to change more code.

Reply:
What if the user wants to load all data from one bucket?

Reply:
The original pr is try to resolve this issue then. If the input path is the bucket name, we should leave the prefix parameter empty in the `ListObject` request, otherwise, s3 will return empty result.

Reply:
```go
	if len(prefix) > 0 && !strings.HasSuffix(prefix, "/") {
```

What about checking suffix too? In case the API is called from other projects.

Reply:
done